ambry-utils/src/test/java/com.github.ambry.utils/SimpleByteBufferPoolTest.java

```diff
@@ -155,6 +155,8 @@ public void testFirstBlockedThenSucceed()
     }
     assertEquals(0, pool.availableMemory());
     consumerThreads[1].start();
+    Thread.sleep(1);
+    assertEquals(Thread.State.TIMED_WAITING, consumerThreads[1].getState());
```

Comment:
This feels like an overkill. Perhaps you could just ensure that allocated[1].getCount() is 1 before used[0].countDown() rather than doing a sleep and checking the thread status.